lib/puppet/util/windows/registry.rb

```diff
@@ -232,9 +242,11 @@ def read(key, name_ptr, *rtype)
         begin
           case type
             when Win32::Registry::REG_SZ, Win32::Registry::REG_EXPAND_SZ
-              result = [ type, data_ptr.read_wide_string(string_length) ]
+              value = data_ptr.read_wide_string(string_length)
+              result = [ type, string_sanitizer.call(value) ]
             when Win32::Registry::REG_MULTI_SZ
-              result = [ type, data_ptr.read_wide_string(string_length).split(/\0/) ]
+              values = data_ptr.read_wide_string(string_length).split(/\0/)
+              result = [ type, values.map!(&string_sanitizer) ]
```

Comment:
Since the string array is already split on NULL with `.split(/\0/)` I don't think `REG_MULTI_SZ` will ever be a problem.  If you really want, you could do this:

```ruby
result = [ type, data_ptr.read_wide_string(string_length).split(/\0/).map!(&:scrub!) ]
```